pkg/machine/hyperv/machine.go

```diff
@@ -170,12 +174,39 @@ func (m *HyperVMachine) Init(opts machine.InitOptions) (bool, error) {
 		Name:      user,
 		Key:       key,
 		VMName:    m.Name,
+		VMType:    machine.HyperVVirt,
 		TimeZone:  opts.TimeZone,
 		WritePath: m.IgnitionFile.GetPath(),
 		UID:       m.UID,
 	}
 
-	if err := machine.NewIgnitionFile(ign, machine.HyperVVirt); err != nil {
+	if err := ign.GenerateIgnitionConfig(); err != nil {
+		return false, err
+	}
+
+	// ready is a unit file that sets up the virtual serial device
+	// where when the VM is done configuring, it will send an ack
+	// so a listening host knows it can being interacting with it
+	//
+	// VSOCK-CONNECT:2 <- shortcut to connect to the hostvm
+	ready := `[Unit]
+After=remove-moby.service sshd.socket sshd.service
+OnFailure=emergency.target
+OnFailureJobMode=isolate
```

Comment:
Can you elaborate on the `isolate` choice?  That means all/most other units are stopped.  Is the idea to nuke everything if this unit fails?

Reply:
maybe a copy and paste from an existing service? I don't remember giving specific instruction to use that, but maybe I did.

The question is, what do you want to happen if this unit fails?

Reply:
it cannot fail

Reply:
then this should work like you want it to. basically the machine won't be up in a half state. it shuts everything down and drops you to a console.